src/taxpasta/infrastructure/application/ganon/ganon_profile.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) 2023 Moritz E. Beber
+# Copyright (c) 2023 Maxime Borry
+# Copyright (c) 2023 James A. Fellows Yates
+# Copyright (c) 2023 Sofia Stamouli.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Provide a description of the ganon profile format."""
+
+
+import numpy as np
+import pandas as pd
+import pandera as pa
+from pandera.typing import Series
+
+from taxpasta.infrastructure.helpers import BaseDataFrameModel
+
+
+class GanonProfile(BaseDataFrameModel):
+    """Define the expected ganon profile format."""
+
+    rank: Series[str] = pa.Field()
+    target: Series[int] = pa.Field(
+        ge=0
+    )  ## TODO: problem - unclassified gets assigned `-`
```

Comment:
My recommendation is that you define this as a `str` column, and then transform it in the standardisation service. Similarly to how it is done in `src/taxpasta/infrastructure/application/metaphlan/metaphlan_profile_standardisation_service.py`. Happy to take over that part.

Reply:
Yes please do that would be nice of you <3, we couldn't work out exactly exactly how/where to do that bit (but @sofstam your intuition was right it seems!)